### PR TITLE
INS-587, INS-585: Remove authorization dummy

### PR DESCRIPTION
**Pull request by ooki2day**
**- What I did**

NetworkCoordinator integrated to service network; authorization on discovery node.

### CODE REVIEW

File: network/hostnetwork/dht.go [hunk 1]
```diff
@@ -1365,3 +1362,70 @@ func (dht *DHT) AddPossibleRelayID(id string) {
 func (dht *DHT) GetOriginHost() *host.Origin {
 	return dht.origin
 }
+
+func (dht *DHT) AddUncheckedNode(hostID id.ID, nonce []byte, ref core.RecordRef) {
+	unchecked := UncheckedNode{Ref: ref, Nonce: nonce}
+	dht.uncheckedNodes[hostID.String()] = unchecked
```
Comment (krestkrest) [inline network/hostnetwork/dht.go]: any time limit to store unchecked nodes? we should clean this map from time to time

File: network/hostnetwork/dht.go [hunk 2]
```diff
@@ -1365,3 +1362,70 @@ func (dht *DHT) AddPossibleRelayID(id string) {
...
+	}
+	sign, err := ecdsa2.Sign(serialized, key)
+	if err != nil {
+		return errors.Wrap(err, "failed to sign a message")
+	}
+	msg.SetSign(sign)
+	return nil
+}
+
+func (dht *DHT) SignedNonceIsCorrect(hostID id.ID, signedNonce []byte) bool {
```
Comment (krestkrest) [inline network/hostnetwork/dht.go]: We could move all methods connected to sign checking to a separate interface, e.g. `SignChecker`, that would be accessible from `HostHandler` interface (via `GetSignChecker()` ). The implementation would have `privateKey` and `uncheckedNodes` as internal state. This step would make `HostHandler` interface lighter.

File: network/hostnetwork/packethandler_test.go [hunk 3]
```diff
@@ -337,6 +366,14 @@ func (hh *mockHostHandler) SetSignChecker(func(msg core.Message) bool) {
 
 }
 
+func (hh *mockHostHandler) SetSigner(func(nonce []byte) []byte) {
```
Comment (AndreyBronin) [inline network/hostnetwork/packethandler_test.go]: define func param as type
type Signer func(nonce []byte) []byte

File: network/hostnetwork/dht.go (hunk 2 above)
Comment (ooki2day) [reply network/hostnetwork/dht.go]: fix: added sign handler

**Pull request merged**